File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []
    for i in range(len(s)):
        if len(stack) == 0:
            stack.append(s[i])
        elif stack[-1] == s[i]:
            stack.pop()
        else:
            stack.append(s[i])
            
    if len(stack) != 0:
        return 0
    return 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.ArrayDeque;
import java.util.Deque;

class Solution
{
    public int solution(String s)
    {
        Deque<String> stack = new ArrayDeque<>();

        for (int i = 0; i < s.length(); i++) {
            if (!stack.isEmpty() && stack.peek().equals(String.valueOf(s.charAt(i)))) {
                stack.pop();
            } else {
                stack.push(String.valueOf(s.charAt(i)));
            }
        }

        if (stack.isEmpty()) {
            return 1;
        } 
        return 0;
    }
}